src/cryptography/hazmat/backends/openssl/rsa.py

```diff
@@ -186,7 +186,14 @@ def _rsa_sig_setup(backend, padding, algorithm, key, data, init_func):
             _Reasons.UNSUPPORTED_HASH,
         )
     res = backend._lib.EVP_PKEY_CTX_set_rsa_padding(pkey_ctx, padding_enum)
-    backend.openssl_assert(res > 0)
+    if res <= 0:
+        backend._consume_errors()
+        raise UnsupportedAlgorithm(
+            "{} is not supported for the RSA sigature operation.".format(
```

Comment:
Typo, signature.

So this path is only reachable if you use `EVP_PKEY_verify_recover_init` and a non-PKCS1v15 padding?

Reply:
Corrected typo.

Yes, before these changes, this code path was protected with an `openssl_assert` only. Errors were not expected to be returned from `EVP_PKEY_CTX_set_rsa_padding since` parameter validation is done in `_rsa_sig_determine_padding`.

Instead of mimicing this approach and checking for valid padding options in the Python layer, I opted for letting the underlying OpenSSL library decide what padding options are valid for signature recovery. Even though we currently know that only PKCS1v15 is accepted, the Python layer doesn't need to know if a padding alg. supports recovery or not (if new padding types are added in the future).